core/dbt/dataclass_schema.py

```diff
@@ -1,97 +1,133 @@
 
-    # This is called by the mashumaro to_dict in order to handle
-    # nested classes.
-    # Munges the dict that's returned.
-    def __post_serialize__(self, dct):
-        if self._hyphenated:
-            new_dict = {}
-            for key in dct:
-                if "_" in key:
-                    new_key = key.replace("_", "-")
-                    new_dict[new_key] = dct[key]
-                else:
-                    new_dict[key] = dct[key]
-            dct = new_dict
-
-        return dct
-
-    # This is called by the mashumaro _from_dict method, before
-    # performing the conversion to a dict
+    # This is called by the mashumaro from_dict in order to handle
+    # nested classes. We no longer do any munging here, but leaving here
+    # so that subclasses can leave super() in place for possible future needs.
     @classmethod
     def __pre_deserialize__(cls, data):
-        # `data` might not be a dict, e.g. for `query_comment`, which accepts
-        # a dict or a string; only snake-case for dict values.
-        if cls._hyphenated and isinstance(data, dict):
-            new_dict = {}
-            for key in data:
-                if "-" in key:
-                    new_key = key.replace("-", "_")
-                    new_dict[new_key] = data[key]
-                else:
-                    new_dict[key] = data[key]
-            data = new_dict
         return data
 
-    # This is used in the hologram._encode_field method, which calls
-    # a 'to_dict' method which does not have the same parameters in
-    # hologram and in mashumaro.
-    def _local_to_dict(self, **kwargs):
-        args = {}
-        if "omit_none" in kwargs:
-            args["omit_none"] = kwargs["omit_none"]
-        return self.to_dict(**args)
+    # This is called by the mashumaro to_dict in order to handle
+    # nested classes. We no longer do any munging here, but leaving here
+    # so that subclasses can leave super() in place for possible future needs.
+    def __post_serialize__(self, data):
+        return data
+
+    @classmethod
+    @functools.lru_cache
+    def json_schema(cls):
+        json_schema_obj = build_json_schema(cls)
+        json_schema = json_schema_obj.to_dict()
+        return json_schema
+
+    @classmethod
+    def clean_dictionary_keys_for_validate(cls, data):
+        return data
+
+    @classmethod
+    def validate(cls, data):
+        data = cls.clean_dictionary_keys_for_validate(data)
+        json_schema = cls.json_schema()
+        validator = jsonschema.Draft7Validator(json_schema)
+        error = next(iter(validator.iter_errors(data)), None)
+        if error is not None:
+            raise ValidationError.create_from(error) from error
+
+    # This method was copied from hologram. Used in model_config.py and relation.py
+    @classmethod
+    def _get_fields(cls) -> List[Tuple[Field, str]]:
+        if cls._mapped_fields is None:
+            cls._mapped_fields = {}
```

Comment:
I think the behavior might be subtly different with initializing to {} instead of None, but maybe not enough to make a difference. This was copied straight from hologram and is only used in a couple of places (which it might be good to revisit), but I'd rather leave this just as it is instead of introducing additional changes in a pull request that's already big.